WordPress/src/main/java/org/wordpress/android/ui/mlp/GutenbergPageLayouts.kt

```diff
@@ -0,0 +1,164 @@
+package org.wordpress.android.ui.mlp
+
+data class GutenbergPageLayouts(
+    val layouts: List<GutenbergLayout>,
+    val categories: List<GutenbergLayoutCategory>
+) {
+    fun layouts(categorySlug: String) =
+            layouts.filter { l -> l.categories.any { c -> c.slug == categorySlug } }
+}
+
+data class GutenbergLayout(
+    val slug: String,
+    val title: String,
+    val preview: String,
+    val categories: List<GutenbergLayoutCategory>
+)
+
+data class GutenbergLayoutCategory(
+    val slug: String,
+    val title: String,
+    val description: String,
+    val emoji: String
+)
+
+/**
+ * Static page layout data factory
+ * NOTE: This implementation will be deprecated and the layouts for self hosted sites will be provided by a new
+ * public API endpoint (/common-block-layouts)
+ */
+object GutenbergPageLayoutFactory {
```

Comment:
@chipsnyder My assumption is that this layout factory implementation will be used only temporarily for testing this PR and the new endpoint @enejb is creating will replace it.